oxo.py

```diff
@@ -1,26 +1,105 @@
+                are currently hard coded because it was nice and easy :(
+                
+            pep-8 is lame so I didn't stick too strictly to it,
+            comments are sparse because doc-strings exist :)
+            
+            UPDATE:
+            I have changed the play.py variable names a lot, because it annoyed
+            me :) but this allowed me to create my OXO game with the potential
+            to have any sized grid and the appropriate checks for them. All
+            grids will be square because rectangular grids make me the big 
+            angery >:( to change the size of the grid, change the variable
+            ' grid_size '  in play.py to any number ( not 1 or 2 they don't 
+            make for a fun game )
+        '''
 
     def get_square(self, x, y):
         """ Return 0, 1 or 2 depending on the contents of the specified square. """
-        raise NotImplementedError("TODO: implement get_square")
+        return self.board[y][x]
 
     def set_square(self, x, y, mark):
         """ If the specified square is currently empty (0), fill it with mark and return True.
             If the square is not empty, leave it as-is and return False. """
-        raise NotImplementedError("TODO: implement set_square")
+        if self.board[y][x] == 0:
+            self.board[y][x] = mark
+            return True
+        else: return False
 
     def is_board_full(self):
         """ If there are still empty squares on the board, return False.
             If there are no empty squares, return True. """
-        raise NotImplementedError("TODO: implement is_board_full")
+        for x in range(self.size):
+            for y in range(self.size):
+                if self.board[y][x] == 0: return False
+        return True
+
+    def horizontal_check(self, mark, index, y):
+        if self.board[y][index] == mark:
+            if index != self.size - 1:
+                return self.horizontal_check(mark, index + 1, y)
+            else: return True
+        else: return False
+
+    def vertical_check(self, mark, x, index):
+        if self.board[index][x] == mark:
+            if index != self.size - 1:
+                return self.vertical_check(mark, x, index + 1)
+            else: return True
+        else: return False
 
     def get_winner(self):
         """ If a player has three in a row, return 1 or 2 depending on which player.
             Otherwise, return 0. """
-        raise NotImplementedError("TODO: implement get_winner")
+
+        # New Horizontal
+        for y in range(self.size):
+            mark = self.board[y][0]
+            if mark != 0:
+                if self.horizontal_check(mark, 1, y):
+                    return mark
+
+        # New Vertical
+        for x in range(self.size):
+            mark = self.board[0][x]
+            if mark != 0:
+                if self.vertical_check(mark, x ,1):
+                    return mark
+
+        # Diagonals
```

Comment:
You defined functions `horizontal_check` and `vertical_check`, so why not `diagonal_check`?